### DISCUSSION TITLE
Explanation of index columns in help

### DISCUSSION THREAD

**Discussion by g0mb4** [neomutt/neomutt#3935] [Development]
Explaining the `$index_format` in the help menu of index could close  #3694.

E.g.:
```
%4C %Z %[%b %d %H:%M] %-15.15L (%<l?%4l&%4c>) %s
```
would result:
```
current message number
a three character set of message status flags
message date/time converted to the local time zone
author name, or recipient name if the message is from you as a list
if number of lines in the message is not 0
  number of lines in the message
else
  number of characters (bytes) in the body of the message
subject of the message
```

**Comment by flatcap** [comment]
More help is good!

But I think just adding more text may not be the best solution...

In #3694, I suggested adding a new help function, but I think overriding `<help>` is smarter.
We could create a new flexible Help Dialog that displays key bindings, plus [lots of other info](https://github.com/neomutt/neomutt/issues/3694#issuecomment-1433071536).

[`parse_version()`](https://github.com/neomutt/neomutt/blob/5bbc865600107986f20066c2e4169ad1866c117d/commands.c#L1478-L1479) is a simple starting point.
Dump some text to a file, then display it in the Pager.

**Comment by flatcap** [comment]
Anyone have any ideas how we could get help for the non-interactive windows: Sidebar, Status Bar, etc?

**Comment by flatcap** [comment]
Steps 1 and 2 of #3937 "Refactoring the Expandos" would allow us to accurately describe the current format string.

**Reply by g0mb4** [reply]
I've created a simple parser for the example string shown above, but it is not as sophisticated as you described.  I'll have some free time, so I can give a try to implement the first step based on your raw notes.

**Reply by flatcap** [reply]
Fantastic!

Unsurprisingly, there are quite a few corner cases :-)
I'll make a list of them all, so we know where we're going.

I think it _should_ be possible to implement steps 1 & 2 for some simple expandos without breaking the complex ones.
